### PR TITLE
Deprecate util functions in public api

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Deprecated utils functions in public api. They will be moved to dev api in 2024.0 release
 - Add SUPPRESS_DEPRECATED macros where these functions are used 

### Tickets:
 - CVS-101941

### CODE REVIEW

File: src/core/include/ngraph/validation_util.hpp [hunk 1]
```diff
@@ -187,6 +202,7 @@ namespace opset1 {
 /// \param      pads_begin        The placeholder for paddings at the beginning of axis.
 /// \param      pads_end          The placeholder for paddings at the end of axis.
 ///
+OPENVINO_DEPRECATED("This function is deprecated and will be moved to dev api in 2024.0 release.")
```
Comment (ilya-lavrenov) [inline src/core/include/ngraph/validation_util.hpp:205]: what are the replacements for all these functions?

Comment (olpipi) [reply src/core/include/ngraph/validation_util.hpp:205]: There are no replacement.
We just plan to move them from public api to dev api in 2024.0

Comment (ilyachur) [reply src/core/include/ngraph/validation_util.hpp:205]: In general you had to deprecate only API from openvino folder... This API anyway will be removed in 2024.0... And we will deprecate all legacy API in the next release. :)

File: src/common/offline_transformations/src/compress_quantize_weigths.cpp [hunk 2]
```diff
@@ -70,7 +70,9 @@ ngraph::pass::CompressQuantizeWeights::CompressQuantizeWeights() {
         auto fq_users = fq->get_users();
         if (fq_users.size() == 1 && has_dequantization_subgraph(fq_users[0])) {
             auto& first_convert = fq_users[0];
+            OPENVINO_SUPPRESS_DEPRECATED_START
             if (auto new_weights = ov::get_constant_from_source(first_convert)) {
+                OPENVINO_SUPPRESS_DEPRECATED_END
```
Comment (ilyachur) [inline src/common/offline_transformations/src/compress_quantize_weigths.cpp:75]: Looks like will be easier just to wrap all if body inside...
```
OPENVINO_SUPPRESS_DEPRECATED_START
if (...) {
} else {
}
OPENVINO_SUPPRESS_DEPRECATED_END
```
In this case we have the same indentation.

File: src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp [hunk 3]
```diff
@@ -206,9 +210,12 @@ bool squeezes_perform_the_same(std::shared_ptr<ov::opset5::Squeeze> lhs, std::sh
         return false;
     const auto l_axes = std::dynamic_pointer_cast<ov::opset5::Constant>(lhs->get_input_node_shared_ptr(1));
     const auto r_axes = std::dynamic_pointer_cast<ov::opset5::Constant>(rhs->get_input_node_shared_ptr(1));
-    if (l_axes && r_axes)
+    if (l_axes && r_axes) {
+        OPENVINO_SUPPRESS_DEPRECATED_START
         return ngraph::normalize_axes(lhs->description(), l_axes->cast_vector<int64_t>(), rank) ==
                ngraph::normalize_axes(rhs->description(), r_axes->cast_vector<int64_t>(), rank);
+    }
+    OPENVINO_SUPPRESS_DEPRECATED_END
```
Comment (ilyachur) [inline src/common/transformations/src/transformations/smart_reshape/strided_slice_squeeze.cpp:218]:
```suggestion
        OPENVINO_SUPPRESS_DEPRECATED_END
    }
```

File: src/core/shape_inference/include/range_shape_inference.hpp [hunk 4]
```diff
@@ -37,7 +37,9 @@ inline bool get_data_as_double<ov::PartialShape>(
     const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data) {
     if (constant_data.count(idx)) {
         axes_value = ov::opset1::Constant(constant_data.at(idx)).cast_vector<double>();
+        OPENVINO_SUPPRESS_DEPRECATED_START
```
Comment (ilyachur) [inline src/core/shape_inference/include/range_shape_inference.hpp:40]: The same idea about if

**Review by ilyachur — APPROVED** [review]
In general LGTM

**Pull request merged**